cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -99,11 +103,18 @@ public void getSubparts(HttpRequest request, HttpResponder responder) throws Bad
   @GET
   @Path("/**/metadata")
   public void getMetadata(HttpRequest request, HttpResponder responder,
-                          @QueryParam("scope") String scope, @QueryParam("type") String type)
+                          @QueryParam("scope") String scope, @QueryParam("type") String type,
+                          @QueryParam("showCustom") boolean showCustom)
     throws BadRequestException {
     MetadataEntity metadataEntity = getMetadataEntityFromPath(request.uri(), type, "/metadata");
-    Set<MetadataRecord> metadata = getMetadata(metadataEntity, scope);
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(metadata, SET_METADATA_RECORD_TYPE));
+    Set<MetadataRecordV2> metadata = getMetadata(metadataEntity, scope);
+    if (showCustom) {
+      // if user has specified to show custom entity resources then no need to make the result compatible
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(metadata, SET_METADATA_RECORD_V2_TYPE));
+    } else {
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(MetadataCompat.filterForCompatibility(metadata),
```

Comment:
In current implementation it will return an empty set but yes you are right we should return 400 since its a bad request. I have made a comment on the issue here to address it and will do it in another PR: 
https://issues.cask.co/browse/CDAP-13260?focusedCommentId=32500&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-32500

Reply:
ok